src/pages/Account/Account.js

```diff
@@ -0,0 +1,247 @@
+import './Account.scss';
+
+const Account = () => {
+  const navigate = useNavigate();
+  const [accountDataName, setAccountDataName] = useState(EMAIL);
+  const [isPwVisible, setIsPwVisible] = useState(false);
+  const [userInfo, setUserInfo] = useState({
+    id: '',
+    pw: '',
+    name: '',
+    tel: '',
+    check: '',
+    birthday: '',
+    domain: '',
+  });
+
+  const { id, pw, name, tel, check, birthday, domain } = userInfo;
+
+  const pwRegExp = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,15})/;
+  const telRegExp = /^([0-9]{3})[-]([0-9]{4})[-][0-9]{4}$/; // 하이픈 포함
+
+  const isNameActive = name.length > 0 || !name;
+  const isTelActive = telRegExp.test(tel) || !tel;
+  const isPwActive = pwRegExp.test(pw) || !pw;
+
+  const onChangeUserInfo = e => {
+    const { name, value } = e.target;
+    setUserInfo({ ...userInfo, [name]: value });
+  };
+
+  const checkDuplicate = e => {
+    // e.preventDefault();
+    fetch('http://10.58.52.223:3000/users/duplicate', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+      },
+      body: JSON.stringify({
+        email: `${id}@${domain}`,
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (!accountDataName === EMAIL) return;
+        if (data.message === 'NEW_USER') {
+          !id || !domain
+            ? alert('이메일을 확인해주세요')
+            : setAccountDataName(SIGNUP);
+        } else {
+          localStorage.setItem('token', data.accessToken);
+          setAccountDataName(LOGIN);
+        }
+      });
+  };
+
+  const checkSignIn = e => {
+    // e.preventDefault();
+    fetch('http://10.58.52.223:3000/users/signin', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+      },
+      body: JSON.stringify({
+        email: `${id}@${domain}`,
+        password: pw,
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        isPwActive && data.accessToken
+          ? navigate('/')
+          : alert('비밀번호를 확인해주세요');
+      });
+  };
+
+  const handlePwVisible = () => {
+    !isPwVisible ? setIsPwVisible(true) : setIsPwVisible(false);
+  };
+
+  const checkSignUp = e => {
+    // e.preventDefault();
+    fetch('http://10.58.52.223:3000/users/signup', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+      },
+      body: JSON.stringify({
+        email: `${id}@${domain}`,
+        password: pw,
+        name: name,
+        phoneNumber: tel,
+        birthday: birthday,
+      }),
+    })
+      .then(response => response.json())
+      .then(() => {
+        if (accountDataName === SIGNUP) {
+          if (
+            isNameActive &&
+            isTelActive &&
+            isPwActive &&
+            name &&
+            tel &&
+            pw &&
+            telRegExp.test(tel) &&
+            pwRegExp.test(pw)
```

Comment:
```suggestion
            isNameActive &&
            isTelActive &&
            isPwActive &&
            name &&
            tel &&
            pw &&
            telRegExp.test(tel) &&
            pwRegExp.test(pw)
```
나중에 꼭! 리팩토링 해주시면 좋을 것 같습니다. 
면접관이 **왜 이렇게 짜셨어요?** 라고 물어보기 딱 좋은 코드거등요 🚀

시간 관계상 이번엔 넘어가도록 하겠습니다 🥲

Reply:
넵..!! 시간이 되면 꼭 리팩토링 하겠습니다 😅